core/storage/btree.rs

```diff
@@ -739,12 +882,23 @@ impl BTreeCursor {
                     BTreeCell::IndexInteriorCell(IndexInteriorCell {
                         left_child_page,
                         payload,
-                        ..
+                        first_overflow_page,
+                        payload_size,
                     }) => {
                         let SeekKey::IndexKey(index_key) = key else {
                             unreachable!("index seek key should be a record");
                         };
-                        let record = crate::storage::sqlite3_ondisk::read_record(payload)?;
+                        let record = if let Some(next_page) = first_overflow_page {
```

Comment:
This code block appears practically identically in at least 6 different places. Should `process_overflow_read()` just be called `process_read_incl_overflow()` and take in `first_overflow_page` (`Option`) and handle both cases:

- When there is no overflow
- When there is overflow

So then you could just call

```
return_if_io!(self.process_read_incl_overflow(payload, first_overflow_page, payload_size));
```

Reply:
I'm scared of making the api to nice because in these cases it is vital that reading from overflow pages requires making sure everything is reentrant

Reply:
I think I simplified it a bit with `CursorResult` now